sdk/communication/AzureCommunicationCalling/Swift_Generated_Interfaces/AzureCommunicationCalling.h

```diff
@@ -0,0 +1,1124 @@
+    open var remoteParticipants: [RemoteParticipant]! { get }
+
+    
+    /// Id of the call
+    open var id: String! { get }
+
+    
+    /// Current state of the call
+    open var state: CallState { get }
+
+    
+    /// Containing code/subcode indicating how a call has ended
+    open var callEndReason: CallEndReason! { get }
+
+    
+    /// Outgoing or Incoming depending on the Call Direction
+    open var callDirection: CallDirection { get }
+
+    
+    /// Whether the local microphone is muted or not.
+    open var isMicrophoneMuted: Bool { get }
+
+    
+    /// Get a list of local video streams in the current call.
+    open var localVideoStreams: [LocalVideoStream]! { get }
+
+    
+    /// Indicates if recording is active in current call
+    open var isRecordingActive: Bool { get }
+
+    
+    /**
+     * The delegate that will handle events from the ACSCall.
+     */
+    unowned(unsafe) open var delegate: CallDelegate!
+
+    
+    /// Mute local microphone.
+    open func mute(completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Unmute local microphone.
+    open func unmute(completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Send DTMF tone
+    open func sendDtmf(tone: DtmfTone, completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Start sharing video stream to the call
+    open func startVideo(stream: LocalVideoStream!, completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Stop sharing video stream to the call
+    open func stopVideo(stream: LocalVideoStream!, completionHandler: ((Error?) -> Void)!)
+
+    
+    /// HangUp a call
+    open func hangUp(options: HangUpOptions!, completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Remove a participant from a call
+    open func remove(participant: RemoteParticipant!, completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Hold this call
+    open func hold(completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Resume this call
+    open func resume(completionHandler: ((Error?) -> Void)!)
+
+    
+    // Class extension begins for Call.
+    open var callerId: CommunicationIdentifier
+
+    open func add(participant: CommunicationIdentifier!) -> RemoteParticipant!
+
+    open func add(participant: PhoneNumberIdentifier!, options: AddPhoneNumberOptions!) -> RemoteParticipant!
+}
+
+// Class extension ends for Call.
+
+
+/// Describes a remote participant on a call
+open class RemoteParticipant : NSObject {
+
+    
+    /// Deallocates the memory occupied by this object.
+    open func dealloc()
+
+    
+    /// Private Preview Only: Display Name of the remote participant
+    open var displayName: String! { get }
+
+    
+    /// True if the remote participant is muted
+    open var isMuted: Bool { get }
+
+    
+    /// True if the remote participant is speaking
+    open var isSpeaking: Bool { get }
+
+    
+    /// Reason why participant left the call, contains code/subcode.
+    open var callEndReason: CallEndReason! { get }
```

Comment:
If I participant hasn't left the meeting, what does this look like? In Swift I might expect it to be nil, but this isn't possible here. Is there a special code for "not ended"?

Reply:
This wont be nil anytime, success of failure will have non-null call end reason. With nullability type specifier support in the future we will fix the need to force un-wrapping.